edb/schema/scalars.py

```diff
@@ -60,7 +62,29 @@ class ScalarType(
     )
 
     sql_type = so.SchemaField(
-        str, default=None, compcoef=0.9)
+        str, default=None, inheritable=False, compcoef=0.9)
+
+    # A type scheme for supporting type mods in scalar types.
+    # If present, describes what the sql_type of children scalars
+    # should be, such as 'varchar({__arg_0__})'.
+    sql_type_scheme = so.SchemaField(
+        str, default=None, inheritable=False, compcoef=0.9)
+
+    # Param definitions for a parameterized type scheme.
+    # Array values notionally say what the type should be,
+    # but currently we only support integers.
+    params = so.SchemaField(
+        checked.FrozenCheckedList[str], default=None,
```

Comment:
Why a `str` though?  Shouldn't this be an `ObjectList[Type]`?

Reply:
Two reasons:
1. Making it `str` was very easy, because all the code to support that exists already. Making it a list of `Type` would require doing some work. We could definitely do it.
2. I wasn't sure if in the future, the params *will* always be edgedb types. They might be "options" or some such?

I don't have a good sense of what all we might be putting there in the future. Looking at postgres, it *might* just end up always being integers, in which case we should just have a *count* and no param types at all.

I can make it be a list of `Type`, though, if you think that's best

Reply:
I think I am going to just make it so that we take a count of parameters.
It's not exposed to introspection, and doesn't need to be backwards compatible, so we can totally rework it in the future, so I'm just going to go with an internal API that is clean and obvious for its current uses.

Reply:
Ok if it's internal